src/Employer/Employer.Web/Services/LegalEntityAgreementService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.EditVacancyInfo;
+
+namespace Esfa.Recruit.Employer.Web.Services
+{
+    public class LegalEntityAgreementService : ILegalEntityAgreementService
+    {
+        private readonly IEmployerVacancyClient _client;
+
+        public LegalEntityAgreementService(IEmployerVacancyClient client)
+        {
+            _client = client;
+        }
+
+        public async Task<bool> HasLegalEntityAgreementAsync(string employerAccountId, long legalEntityId)
+        {
+            var legalEntity = await GetLegalEntityAsync(employerAccountId, legalEntityId);
+
+            if(legalEntity == null)
+                return false;
+            
+            if (legalEntity.HasLegalEntityAgreement)
+                return true;
+
+            //Agreement may have been signed since the projection was created. Check Employer Service.
```

Comment:
Don't know if this is possible but could there be a case where a contract is superseded by another and is yet to be signed?

Reply:
Yep this could happen. However at the moment we only need to check that at least one contract is signed regardless of the version. I've checked this with @mgwilliam